package.json

```diff
@@ -137,11 +137,11 @@
   "scripts": {
     "clean": "npm run clean:local --workspaces",
     "compile": "npm run compile:local --workspaces",
-    "build": "rm -rf dist/ && npm run lint && npm run compile:local && npm run generate-types && npm run compile-webpack-loader",
-    "clean:local": "echo Nothing to clean",
-    "compile:local": "rollup --config rollup/rollup.config.ts --configPlugin 'typescript={tsconfig: `tsconfig.rollup.json`}' --vscode-version ${npm_package_config_vscode_version}",
+    "build": "npm run clean:local && npm run compile:local && npm run lint && npm run generate-types && npm run compile-webpack-loader",
+    "clean:local": "rm -rf dist/",
+    "compile:local": "npm run prepare:vscode && rollup --config rollup/rollup.config.ts --configPlugin 'typescript={tsconfig: `tsconfig.rollup.json`}' --vscode-version ${npm_package_config_vscode_version}",
     "compile-webpack-loader": "tsc --target es2020 -lib es2020 --module commonjs --skipLibCheck --moduleResolution node --declaration --outDir dist src/webpack-loader.ts && mv dist/webpack-loader.js dist/webpack-loader.cjs",
-    "preprepare": "./scripts/install-vscode",
+    "prepare:vscode": "./scripts/install-vscode",
```

Comment:
Why do we need this change?

Reply:
It bugs me that `npm i` always run the vscode installation, but my proposed solution is not good. I will come up with a better idea tomorrow.

Do you think it is a good idea to have one workspace being a sub-directory of another workspace (contained in I mean)? So far, I only have created workspace that don't overlap.

Reply:
> I will come up with a better idea tomorrow.

I agree a kind a cache could be nice! we can probably skip the script if the vscodeRef didn't change

> Do you think it is a good idea to have one workspace being a sub-directory of another workspace (contained in I mean)? So far, I only have created workspace that don't overlap.

I won't say I'm an expert of npm workspaces, they are a very hard thing to understand and master and that was the reason I didn't use it at first so I don't really know 🤷

Reply:
> I agree a kind a cache could be nice! we can probably skip the script if the vscodeRef didn't change

I have build this in now ⬇️ 

> I won't say I'm an expert of npm workspaces, they are a very hard thing to understand and master and that was the reason I didn't use it at first so I don't really know 🤷

It works for now. Longer term, I would suggest to move the main package into its own sub dir like:
```shell
api
demo
```

Reply:
It bothers me a little to have to move the main package into a subdirectory just to be able to add a little demo project next to it

Reply:
yes, that's why I haven't done it, so far. Maybe it is a better idea to exclude it from the package and use vite and separate tsconfig for the demo and move it to the main source folder. WDYT?

Reply:
Should I build a POC on a separate commit?

Reply:
It looks like a good idea!

Reply:
Discussion is outdated.